packages/mobile/src/verify/VerificationLoadingScreen.tsx

```diff
@@ -1,34 +1,110 @@
 import colors from '@celo/react-components/styles/colors'
+import { fontStyles } from '@celo/react-components/styles/fonts'
 import * as React from 'react'
 import { withNamespaces, WithNamespaces } from 'react-i18next'
-import { ScrollView, StyleSheet } from 'react-native'
+import { ScrollView, StyleSheet, Text, View } from 'react-native'
 import SafeAreaView from 'react-native-safe-area-view'
+import { connect } from 'react-redux'
+import { hideAlert, showError } from 'src/alert/actions'
+import { errorSelector } from 'src/alert/reducer'
 import componentWithAnalytics from 'src/analytics/wrapper'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import CancelButton from 'src/components/CancelButton'
 import Carousel, { CarouselItem } from 'src/components/Carousel'
 import DevSkipButton from 'src/components/DevSkipButton'
 import { Namespaces } from 'src/i18n'
+import LoadingSpinner from 'src/icons/LoadingSpinner'
+import NuxLogo from 'src/icons/NuxLogo'
+import {
+  cancelVerification,
+  receiveAttestationMessage,
+  startVerification,
+} from 'src/identity/actions'
+import { AttestationCode } from 'src/identity/verification'
 import { Screens } from 'src/navigator/Screens'
+import { RootState } from 'src/redux/reducers'
+import { currentAccountSelector } from 'src/web3/selectors'
 
-class VerificationLoadingScreen extends React.Component<WithNamespaces> {
-  static navigationOptions = null
+interface StateProps {
+  numberVerified: boolean
+  e164Number: string
+  account: string | null
+  attestationCodes: AttestationCode[]
+  numCompleteAttestations: number
+  verificationFailed: boolean
+  underlyingError: ErrorMessages | null | undefined
+}
+
+interface DispatchProps {
+  startVerification: typeof startVerification
+  cancelVerification: typeof cancelVerification
+  receiveVerificationMessage: typeof receiveAttestationMessage
+  showError: typeof showError
+  hideAlert: typeof hideAlert
+}
+
+type Props = StateProps & DispatchProps & WithNamespaces
+
+const mapDispatchToProps = {
+  startVerification,
+  cancelVerification,
+  receiveVerificationMessage: receiveAttestationMessage,
+  showError,
+  hideAlert,
+}
+
+const mapStateToProps = (state: RootState): StateProps => {
+  return {
+    numberVerified: state.app.numberVerified,
+    e164Number: state.account.e164PhoneNumber,
+    attestationCodes: state.identity.attestationCodes,
+    numCompleteAttestations: state.identity.numCompleteAttestations,
+    verificationFailed: state.identity.verificationFailed,
+    account: currentAccountSelector(state),
+    underlyingError: errorSelector(state),
+  }
+}
+
+class VerificationLoadingScreen extends React.Component<Props> {
+  static navigationOptions = { header: null }
+
+  onCancel = () => {
```

Comment:
what should happen here?

Reply:
We cancel the verification saga. Will do in next PR